### ISSUE TITLE
error NU1101: Unable to find package dotnet-xunit. No packages exist with this id in source(s)

### ISSUE THREAD

**Issue by CreepyGnome** [xunit/xunit#1427]
Trying to run xUnit tests in VSTS (visualstudio.com) in a dotnet build task and it fails on restore with:

> error NU1101: Unable to find package dotnet-xunit. No packages exist with this id in source(s): D:\VSTS-BuildAgent\_work\_tool\dncs\2.0.0\x64\sdk\NuGetFallbackFolder

Everything works from my box from the command line and XML result file occurs. Restore works and everything.  I have the csproj files modified per the docs I have followed here: https://xunit.github.io/docs/getting-started-dotnet-core

Here is the error and stack trace from the VSTS build log:

```
2017-08-25T01:17:53.4103655Z 18:17:53.410     1>D:\VSTS-BuildAgent\_work\5\s\src\Tests\###CENSORED###\###CENSORED###.csproj : error NU1101: Unable to find package dotnet-xunit. No packages exist with this id in source(s): D:\VSTS-BuildAgent\_work\_tool\dncs\2.0.0\x64\sdk\NuGetFallbackFolder
2017-08-25T01:17:53.4193717Z 18:17:53.419     1>D:\VSTS-BuildAgent\_work\_tool\dncs\2.0.0\x64\sdk\2.0.0\NuGet.targets(102,5): error : Value cannot be null. [D:\VSTS-BuildAgent\_work\5\s\src\Tests\###CENSORED###\###CENSORED###.csproj]
2017-08-25T01:17:53.4193717Z D:\VSTS-BuildAgent\_work\_tool\dncs\2.0.0\x64\sdk\2.0.0\NuGet.targets(102,5): error : Parameter name: path [D:\VSTS-BuildAgent\_work\5\s\src\Tests\###CENSORED###\###CENSORED###.csproj]
2017-08-25T01:17:53.4493971Z                      System.ArgumentNullException: Value cannot be null.
2017-08-25T01:17:53.4493971Z                      Parameter name: path
2017-08-25T01:17:53.4493971Z                         at System.IO.Path.GetFullPath(String path)
2017-08-25T01:17:53.4493971Z                         at NuGet.Commands.RestoreCommand.GetAssetsFilePath(LockFile lockFile)
2017-08-25T01:17:53.4493971Z                         at NuGet.Commands.RestoreCommand.<ExecuteAsync>d__8.MoveNext()
2017-08-25T01:17:53.4493971Z                      --- End of stack trace from previous location where exception was thrown ---
2017-08-25T01:17:53.4493971Z                         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-08-25T01:17:53.4493971Z                         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-08-25T01:17:53.4493971Z                         at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
2017-08-25T01:17:53.4493971Z                         at NuGet.Commands.RestoreRunner.<ExecuteAsync>d__7.MoveNext()
2017-08-25T01:17:53.4493971Z                      --- End of stack trace from previous location where exception was thrown ---
2017-08-25T01:17:53.4493971Z                         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-08-25T01:17:53.4493971Z                         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-08-25T01:17:53.4493971Z                         at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
2017-08-25T01:17:53.4493971Z                         at NuGet.Commands.RestoreRunner.<ExecuteAndCommitAsync>d__6.MoveNext()
2017-08-25T01:17:53.4493971Z                      --- End of stack trace from previous location where exception was thrown ---
2017-08-25T01:17:53.4493971Z                         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-08-25T01:17:53.4493971Z                         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-08-25T01:17:53.4493971Z                         at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
2017-08-25T01:17:53.4493971Z                         at NuGet.Commands.RestoreRunner.<CompleteTaskAsync>d__9.MoveNext()
2017-08-25T01:17:53.4493971Z                      --- End of stack trace from previous location where exception was thrown ---
2017-08-25T01:17:53.4493971Z                         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-08-25T01:17:53.4503799Z                         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-08-25T01:17:53.4503799Z                         at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
2017-08-25T01:17:53.4503799Z                         at NuGet.Commands.RestoreRunner.<RunAsync>d__2.MoveNext()
2017-08-25T01:17:53.4503799Z                      --- End of stack trace from previous location where exception was thrown ---
2017-08-25T01:17:53.4503799Z                         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-08-25T01:17:53.4503799Z                         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-08-25T01:17:53.4503799Z                         at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
2017-08-25T01:17:53.4503799Z                         at NuGet.Commands.RestoreRunner.<RunAsync>d__0.MoveNext()
2017-08-25T01:17:53.4503799Z                      --- End of stack trace from previous location where exception was thrown ---
2017-08-25T01:17:53.4503799Z                         at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
2017-08-25T01:17:53.4503799Z                         at System.Runtime.CompilerServices.TaskAwaiter.HandleNonSuccessAndDebuggerNotification(Task task)
2017-08-25T01:17:53.4503799Z                         at System.Runtime.CompilerServices.TaskAwaiter`1.GetResult()
2017-08-25T01:17:53.4503799Z                         at NuGet.Build.Tasks.RestoreTask.<ExecuteAsync>d__31.MoveNext() (TaskId:43)
```

**Comment by bradwilson** [comment]
Why are you reporting this to us? This is either a VSTS or NuGet bug. We don't implement package restore.

**Comment by CreepyGnome** [comment]
Sorry I figured you guys would be the first to know whats wrong since I followed your documentation and it is erroring on your package. So who do I resolve an issue when I follow your docs and they don't work?

**Comment by bradwilson** [comment]
You can leave the issue open if you want, but since this appears to be some kind of networking issue in either VSTS or an issue with the NuGet client, there's not going to be anything *we* will be able to do about it.

**Comment by CreepyGnome** [comment]
Related issue I provide this error as well I get when it tries to get the package dotnet-xunit:

`Package 'dotnet-xunit 2.3.0-beta4-build3742' has a package type 'DotnetCliTool' that is not supported by project 'Tests\SM.Common.Tests'.`

Is this also a NuGet issue?  I have the correct project type per your documentation.

`<TargetFramework>netcoreapp2.0</TargetFramework>`

Also when you run

`dotnet new classlib`

On dotnet 2.0 it creates a DotNet Standard project that has this:

`<TargetFramework>netstandard2.0</TargetFramework>`

Is that what we need and your docs don't work with netcoreapp2.0?

However when I follow your docs step by step to the letter from command line and I don't change the framework to match your docks leaving it as netstandard2.0 I get this error when I run donet xunit:

```
WebApplication2\MyFirstUnitTests\bin\Debug\netstandard2.0\MyFirstUnitTests.dll
Unsupported target framework '.NETStandard 2.0' (only .NETCoreApp 1.0+ and .NETFramework 4.5.2+ are supported)
```

From my POV using your docs it feels like its needs some work on the extension and ensuring it works with VSTS I guess.

**Comment by bradwilson** [comment]
No, the docs explicitly call out that you must use `netcoreapp`:

> We've made two important changes here:
> * We changed the target framework to `netcoreapp2.0`, which means that our tests will use .NET Core 2.0.x (whatever is the latest version installed on our machine);
> * We added a new package reference to `xunit`, as well as a new .NET CLI Tool reference to `dotnet-xunit`. The former is the unit testing framework, and the latter is the command line tooling that allows you to run the unit tests.

@CreepyGnome writes:

> Is this also a NuGet issue?

Are you trying to use the NuGet UI? If so, then this is a known limitation of that UI (it does not support .NET SDK CLI packages).

**Comment by CreepyGnome** [comment]
I don't think you understand following your docs to the letter cause one of the issues then if I change it to netcoreapp2.0 it causes another issue.  Basically the docs are not working.

I have used the dotnet CLI for recreating the issue. I tried to use the NuGet VS2017 UI as well and it cannot add dotnet-xunit with it.  I have manually added the CSProj and even with the CSProj I get errors on my local box attempting to reproduce the VSTS issues.  Even with all these NuGet errors when I run dotnet xunit it works locally. However since there were errors VSTS just stops.

So in VSTS its VS2017 Build host is 2017 Update 1 (15.1) so I added the task to install DotNet 2.0 SDK before it resetores and it still doesn't work.   

So I understand VSTS for some reason is two VS2017 updates behind, but something else in the updates seems to be required as I do force 2.0 SDK to be installed which doesn't resolve something that Update 3 probably does resolve.

So I will probably have to wait for VSTS to update their hosted vs2017 agent. Which means while I can ignore the errors locally I cannot have the changes checked in as it will break VSTS. SO I will have to wait for Microsoft to update their stuff to try again.

Thanks @bradwilson for your patience on this, I do appreciate your time trying to help me.
